packages/aws-rfdk/lib/lambdas/nodejs/lib/deadline-client/test/deadline-client.test.ts

```diff
@@ -0,0 +1,360 @@
+        port: 8080,
+        protocol: 'HTTP',
+      });
+
+      const promise = deadlineClient.GetRequest('/get/version/test');
+      response.emit('data', Buffer.from(JSON.stringify(''), 'utf8'));
+      response.emit('end');
+      promise.then(resp => resp).catch(err => err);
+
+      // THEN
+      // should make an HTTP request
+      expect(jest.requireMock('http').request)
+        .toBeCalledWith(
+          {
+            agent: undefined,
+            method: 'GET',
+            port: 8080,
+            host: 'hostname',
+            path: '/get/version/test',
+          },
+          expect.any(Function),
+        );
+    });
+
+    test('successful http get request with options', async () => {
+      // GIVEN
+      jest.requireMock('http').request.mockImplementation(httpRequestMock);
+      response = new MockResponse();
+
+      // WHEN
+      deadlineClient = new DeadlineClient({
+        host: 'hostname',
+        port: 8080,
+        protocol: 'HTTP',
+      });
+
+      const promise = deadlineClient.GetRequest('/get/version/test', {
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      });
+      response.emit('data', Buffer.from(JSON.stringify(''), 'utf8'));
+      response.emit('end');
+      promise.then(resp => resp).catch(err => err);
+
+      // THEN
+      // should make an HTTP request
+      expect(jest.requireMock('http').request)
+        .toBeCalledWith(
+          {
+            agent: undefined,
+            method: 'GET',
+            port: 8080,
+            host: 'hostname',
+            path: '/get/version/test',
+          },
+          expect.any(Function),
+        );
+    });
+
+    test('failed http get request', async () => {
+      // GIVEN
+      response = new MockResponse();
+      response.statusCode = 400;
+      jest.requireMock('http').request.mockImplementation(httpRequestMock);
+
+      // WHEN
+      deadlineClient = new DeadlineClient({
+        host: 'hostname',
+        port: 8080,
+        protocol: 'HTTP',
+      });
+      const promise = deadlineClient.GetRequest('/get/version/test');
+
+      // THEN
+      await expect(promise)
+        .rejects
+        .toThrowError(expect.any(Error));
+    });
+
+    test('successful https get request', async () => {
+      // GIVEN
+      jest.requireMock('https').request.mockImplementation(httpRequestMock);
+      response = new MockResponse();
+
+      // WHEN
+      deadlineClient = new DeadlineClient({
+        host: 'hostname',
+        port: 4433,
+        protocol: 'HTTPS',
+      });
+
+      const promise = deadlineClient.GetRequest('/get/version/test');
+      response.emit('data', Buffer.from(JSON.stringify(''), 'utf8'));
+      response.emit('end');
+      promise.then(resp => resp).catch(err => err);
+
+      // THEN
+      // should make an HTTP request
+      expect(jest.requireMock('https').request)
+        .toBeCalledWith(
+          {
+            agent: expect.any(Object),
```

Comment:
we can tighten this validation a bit with:

```ts
// THEN
const agentMock = jest.requireMock('https').Agent as jest.Mock;
expect(agentMock).toHaveBeenCalledTimes(1)
expect(agentMock).toBeCalledWith(
  expect.objectContaining({
    ca: 'content',
    pfx: 'content',
    passphrase: 'content',
  }),
);

// should make an HTTP request
expect(jest.requireMock('https').request)
  .toBeCalledWith(
    {
      agent: agentMock.mock.instances[0],
      method: 'GET',
      port: 4433,
      host: 'hostname',
      path: '/get/version/test',
    },
    expect.any(Function),
  );
```

For this to work, we need to be sure to clear the Agent mock's collected instances and calls between tests with:

```ts
beforeEach(() => {
  (jest.requireMock('https').Agent as jest.Mock).mockClear();
});
```

Reply:
That's much better. Thanks!